app/controllers/groups_controller.rb

```diff
@@ -715,15 +715,15 @@ def test_email_settings
     params.require(:host)
     params.require(:username)
     params.require(:password)
-    params.require(:ssl)
+
+    raise Discourse::InvalidParameters if params[:ssl_mode].blank? && params[:ssl].blank?
```

Comment:
That confused me a bit as `smtp` is using new `:ssl_mode` and `imap` is using `ssl`. 

What do you think about moving the raise to a specific branches:
```ruby
      begin
        case params[:protocol]
        when "smtp"
           raise Discourse::InvalidParameters if params[:ssl_mode].blank?
           ...
        when "imap"
          raise Discourse::InvalidParameters ifparams[:ssl].blank?
```

Reply:
Oh yeah good point. I didn't focus as much on IMAP since we are going to blow it up anyway lol. But I will change this :+1: